### PR TITLE
20165-Support-segment-path-printing

### PR DESCRIPTION
**Pull request by pavel-krivanek**
https://pharo.fogbugz.com/f/cases/20165/Support-segment-path-printing

### CODE REVIEW

File: src/FileSystem-Core.package/Path.class/class/from.delimiter..st [hunk 1]
```diff
@@ -5,8 +5,10 @@ from: aString delimiter: aDelimiterCharacter
...
 	
-	pathClass :=  ((self isAbsoluteUnixPath: aString) or: [self isAbsoluteWindowsPath: aString]) 
+	pathClass :=  ((self isAbsolutePath: aString delimiter: aDelimiterCharacter) or: 
+							[self isAbsoluteWindowsPath: aString]) 
 		ifTrue: [ AbsolutePath ]
 		ifFalse:[ RelativePath ].
 	
-	^ pathClass withAll: (pathClass canonicalizeElements: (aDelimiterCharacter split: aString))
\ No newline at end of file
+	^ pathClass withAll: ((aDelimiterCharacter split: aString) select:
```
Comment (gcotelli) [inline src/FileSystem-Core.package/Path.class/class/from.delimiter..st]: Why canonicalizeElements is not needed anymore?

Comment (akgrant43) [inline src/FileSystem-Core.package/Path.class/class/from.delimiter..st]: Hi @gcotelli,

Thanks for reviewing this, and my apologies for the slow reply (I wasn't subscribed to the issue :-().

Just FYI, the justification for removing the canonicalisation is described in:
http://forum.world.st/FileReference-gt-gt-and-path-canonicalisation-td4952575.html

However it isn't part of this patch.  I obviously forgot to check that I wasn't modifying a method that had already been modified as part of my other work.

Great catch, and thanks for letting me know.

Pavel, my apologies also, but I think you need to make this change as you own the branch that generated the pull request.  If there is a way I can do it, please let me know so I can save time in future.

I'll make the change, test it, and post the modified method in a follow-up comment (real soon now).

Thanks again,
Alistair

Comment (akgrant43) [inline src/FileSystem-Core.package/Path.class/class/from.delimiter..st]: And here's the modified method:

from: aString delimiter: aDelimiterCharacter 
	"Answer a path composed of several elements delimited by aCharacter"
	| pathClass |
	aString isEmpty
		ifTrue: [ ^ self root ].
	
	pathClass :=  ((self isAbsolutePath: aString delimiter: aDelimiterCharacter) or: 
							[self isAbsoluteWindowsPath: aString]) 
		ifTrue: [ AbsolutePath ]
		ifFalse:[ RelativePath ].
	
	^ pathClass withAll: (pathClass canonicalizeElements: (aDelimiterCharacter split: aString))

**Comment by akgrant43** [comment]
Hi Pavel,

This PR is significantly broken...

It seems to think that Path and AbsolutePath are part of the FileSystem-Core package, when they are in fact part of the FileSystem-Path package.  The causes weird errors when trying to load the associated branch from Iceberg.

I've created a new pull request with the same fixes, and fixing the merge conflicts caused by this version, in #210.

Cheers,
Alistair

**Comment by pavel-krivanek** [comment]
So should we close this one?

**Comment by akgrant43** [comment]
Hi Pavel,

> So should we close this one?

Yes! (and maybe delete the branch so no-one accidentally looks at it later on) :-)

Thanks,
Alistair

**Pull request closed without merging**